class.js

```diff
@@ -0,0 +1,405 @@
+        material.defines = this.defines;
+        let db = {
+            name: `u_doubleBuffer${index}`,
+            material: material,
+            renderTargets: [],
+            width: width,
+            height: height,
+            wrapS: RepeatWrapping,
+            wrapT: RepeatWrapping,
+            minFilter: LinearFilter,
+            magFilter: LinearFilter
+        };
+
+        this.doubleBuffers.push(db);
+        this.uniforms[db.name] = { value: null };
+
+        db.renderTargets[0] = this.createRenderTarget(db);
+        db.renderTargets[1] = this.createRenderTarget(db);
+
+        return db;
+    }
+
+    addPostprocessing(frag_src) {
+        this.postprocessing = createShaderMaterial(`#define POSTPROCESSING\n${frag_src}`, this.uniforms);
+        this.postprocessing.defines = this.defines;
+
+        this.sceneBuffer = {
+            renderTarget: null,
+            width: this.renderer.domElement.width,
+            height: this.renderer.domElement.height,
+        };
+
+        this.uniforms["u_scene"] = { value: null };
+
+        this.sceneBuffer.renderTarget = this.createRenderTarget({
+            width: this.sceneBuffer.width,
+            height: this.sceneBuffer.height,
+            wrapS: null,
+            wrapT: null,
+            minFilter: LinearFilter,
+            magFilter: LinearFilter
+        });
+
+        return this.sceneBuffer;
+    }
+
+    createRenderTarget(b) {
+        b.wrapS = b.wrapS || ClampToEdgeWrapping;
+        b.wrapT = b.wrapT || ClampToEdgeWrapping;
+
+        b.minFilter = b.minFilter || NearestFilter;
+        b.magFilter = b.magFilter || NearestFilter;
+
+        let type = FloatType;
+
+        if (this.renderer.capabilities.isWebGL2 === false)
+            type = HalfFloatType;
+
+        let w = b.width;
+        let h = b.height;
+
+        if (w <= 1.0 && h <= 1.0) {
+            w *= this.renderer.domElement.width;
+            h *= this.renderer.domElement.height;
+        }
+
+        let renderTarget = new WebGLRenderTarget(Math.floor(w), Math.floor(h), {
+            wrapS: b.wrapS,
+            wrapT: b.wrapT,
+            minFilter: b.minFilter,
+            magFilter: b.magFilter,
+            format: RGBAFormat,
+            type: (/(iPad|iPhone|iPod)/g.test(navigator.userAgent)) ? HalfFloatType : type,
+            stencilBuffer: false
+        });
+
+        return renderTarget;
+    }
+
+    updateUniforms() {
+        this.time = this.clock.getElapsedTime();
+
+        this.uniforms.u_time.value = this.time;
+        this.uniforms.u_delta.value = this.time - this.lastTime;
+        this.uniforms.u_frame.value = this.frame;
+
+        this.lastTime = this.time;
+        this.frame++;
+    }
+
+    updateBuffers() {
+        // Buffers
+        for (let i = 0, il = this.buffers.length; i < il; i++) {
+            let b = this.buffers[i];
+            if (db.width <= 1.0 && db.height <= 1.0)
```

Comment:
There is a fix that needs to be update here https://github.com/patriciogonzalezvivo/glsl-sandbox/commit/f3bfc7c2cd9a30df7da9672418847a4a02bc232d